model/end-device-lorawan-mac.cc

```diff
@@ -87,7 +87,7 @@ EndDeviceLorawanMac::GetTypeId (void)
                      "ns3::TracedValueCallback::Double")
     .AddAttribute ("MaxTransmissions",
                    "Maximum number of transmissions for a packet",
-                   IntegerValue (8),
+                   IntegerValue (1),
                    MakeIntegerAccessor (&EndDeviceLorawanMac::m_maxNumbTx),
                    MakeIntegerChecker<uint8_t> ())
```

Comment:
You can also use `MakeIntegerChecker`'s arguments to specify minimum and maximum allowed values:
```suggestion
                   MakeIntegerChecker<uint8_t> (1, 15))
```

Reply:
I made this change, but no error is given when I run complete-network-example with numberOfTransmissions=17? I tried googling this checker but didn't get examples of how this should work.

Reply:
This is expected, as the integer value is checked only if the value is set through the attribute system, which is bypassed if you use the method directly!